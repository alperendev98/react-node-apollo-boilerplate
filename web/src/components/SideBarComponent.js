import React from 'react';
import { getBEMClasses } from '../helpers/cssClassesHelper';

import logoIcon from '../assets/images/logo.svg';
import invoiceIcon from '../assets/images/icon-invoice.svg';
import mailIcon from '../assets/images/icon-mail.svg';
import settingIcon from '../assets/images/icon-settings.svg';
import speedIcon from '../assets/images/icon-speed.svg';
import targetIcon from '../assets/images/icon-target.svg';

import '../styles/base/base.css';
import '../styles/components/sidebarComponent.css';

const BaseClasses = getBEMClasses(['base']);
const SideBarClasses = getBEMClasses(['side-bar']);

class SideBarComponent extends React.Component {
	render() {
		const { showSideNav, hideSideNav } = this.props;

		return (
			<div className={SideBarClasses('container')}>
				<div className={SideBarClasses('logo')}>
					<img className={SideBarClasses('logo-icon')} src={logoIcon} />
				</div>
				<div className={SideBarClasses('nav')}>
					<div className={SideBarClasses('nav-item')}>
						<img className={SideBarClasses('nav-item-icon')} src={invoiceIcon} />
					</div>
					<div
						onMouseEnter={showSideNav}
						onMouseLeave={hideSideNav}
						className={SideBarClasses('nav-item', 'manage')}
					>
						<img className={SideBarClasses('nav-item-icon')} src={speedIcon} />
					</div>
					<div className={SideBarClasses('nav-item')}>
						<img className={SideBarClasses('nav-item-icon')} src={targetIcon} />
					</div>
					<div className={SideBarClasses('nav-item')}>
						<img className={SideBarClasses('nav-item-icon')} src={mailIcon} />
					</div>
					<div className={SideBarClasses('nav-item')}>
						<img className={SideBarClasses('nav-item-icon')} src={settingIcon} />
					</div>
				</div>
			</div>
		);
	}
}

export default SideBarComponent;
