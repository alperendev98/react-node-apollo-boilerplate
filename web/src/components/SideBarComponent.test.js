import React from 'react';
import { mount, shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import SideBarComponent from './SideBarComponent';

const mockStore = configureStore();

describe('SideBarComponent', () => {
	const initialState = {};
	const initialProps = {
		showSideNav: jest.fn(),
		hideSideNav: jest.fn()
	};
	let component, wrapper, store;

	beforeEach(() => {
		store = mockStore(initialState);
		component = mount(<SideBarComponent {...initialProps} store={store} />, {});
		wrapper = shallow(<SideBarComponent />);
	});

	it('shallowly renders SideBarComponent without crashing', () => {
		expect(wrapper.length).toEqual(1);
	});

	it('should render SideBarComponent with .side-bar__container class', () => {
		expect(component.find('.side-bar__container').length).toBe(1);
	});

	it('should call props.showSideNav with mouse enters manage icon', () => {
		const props = component.props();
		jest.spyOn(props, 'showSideNav');
		component.find('.side-bar__nav-item--manage').simulate('mouseenter');
		expect(props.showSideNav).toHaveBeenCalled();
	});

	it('should call props.hideSideNav when mouse leaves manage icon', () => {
		const props = component.props();
		jest.spyOn(props, 'hideSideNav');
		component.find('.side-bar__nav-item--manage').simulate('mouseenter');
		component.find('.side-bar__nav-item--manage').simulate('mouseleave');
		expect(props.hideSideNav).toHaveBeenCalled();
	});

	it('should match previos snapshot for SideBarComponent', () => {
		expect(wrapper).toMatchSnapshot();
	});
});
