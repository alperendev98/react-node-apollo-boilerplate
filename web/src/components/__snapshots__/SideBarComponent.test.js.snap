// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideBarComponent should match previos snapshot for SideBarComponent 1`] = `
<div
  className="side-bar__container"
>
  <div
    className="side-bar__logo"
  >
    <img
      className="side-bar__logo-icon"
      src="logo.svg"
    />
  </div>
  <div
    className="side-bar__nav"
  >
    <div
      className="side-bar__nav-item"
    >
      <img
        className="side-bar__nav-item-icon"
        src="icon-invoice.svg"
      />
    </div>
    <div
      className="side-bar__nav-item side-bar__nav-item--manage"
    >
      <img
        className="side-bar__nav-item-icon"
        src="icon-speed.svg"
      />
    </div>
    <div
      className="side-bar__nav-item"
    >
      <img
        className="side-bar__nav-item-icon"
        src="icon-target.svg"
      />
    </div>
    <div
      className="side-bar__nav-item"
    >
      <img
        className="side-bar__nav-item-icon"
        src="icon-mail.svg"
      />
    </div>
    <div
      className="side-bar__nav-item"
    >
      <img
        className="side-bar__nav-item-icon"
        src="icon-settings.svg"
      />
    </div>
  </div>
</div>
`;
